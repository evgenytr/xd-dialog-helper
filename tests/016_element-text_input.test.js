const textInput = require('../src/main').types.TEXT_INPUT;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {}),
    close: jest.fn(() => {}),
    cancel: jest.fn(() => {}),
    registerElement: jest.fn(() => undefined)
};

let rendered;

describe('Text Input Element', () => {
    beforeEach(() => {
        rendered = textInput.render('dialog', {
            id: 'hr',
            label: 'Hello World',
            value: 'abc',
            htmlAttributes: {
                someAttr: 'hello'
            }
        }, actions);
    });

    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();

            expect(textInput.render('dialog',{id: 'noHTML'}, actions)).toMatchSnapshot();
        });

        it('should correctly handle changing values', () => {
            rendered.input.value = '6';
            rendered.input.dispatchEvent(new Event('input'));

            expect(actions.change).toBeCalled();
            expect(textInput.value(rendered)).toBe('6');
        });
    });
    describe('value()', () => {
        it('should correctly return the value', () => {
            expect(textInput.value(rendered)).toBe('abc');
        });
    });
    describe('validate()', () => {
        it('should always return true if there\'s no required prop', () => {
            expect(textInput.valid(rendered)).toBeTruthy();
        });

        it('should return false for empty fields if required == true', () => {
            let emptyField = textInput.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: '',
                required: true
            }, actions);
            expect(textInput.valid(emptyField)).toBeFalsy();
        });

        it('should return true for non-empty fields if required == true', () => {
            let emptyField = textInput.render('dialog',  {
                id: 'helloworld',
                label: 'My world',
                value: 'a',
                required: true
            }, actions);
            expect(textInput.valid(emptyField)).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(textInput.type).toMatchSnapshot();
    });
});
