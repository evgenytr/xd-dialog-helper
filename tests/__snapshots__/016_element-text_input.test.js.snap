// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Input"`;

exports[`Text Input Element render() should render correctly 1`] = `
Object {
  "input": <input
    id="dialog-hr"
    placeholder="Hello World"
    someattr="hello"
  />,
  "props": Object {
    "htmlAttributes": Object {
      "someAttr": "hello",
    },
    "id": "hr",
    "label": "Hello World",
    "value": "abc",
  },
  "type": Object {
    "render": [Function],
    "type": "Input",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-hr-wrapper"
  >
    <span
      id="dialog-hr-label"
    >
      Hello World
    </span>
    <input
      id="dialog-hr"
      placeholder="Hello World"
      someattr="hello"
    />
  </label>,
}
`;

exports[`Text Input Element render() should render correctly 2`] = `
Object {
  "input": <input
    id="dialog-noHTML"
    placeholder="undefined"
  />,
  "props": Object {
    "id": "noHTML",
  },
  "type": Object {
    "render": [Function],
    "type": "Input",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <label
    id="dialog-noHTML-wrapper"
  >
    <span
      id="dialog-noHTML-label"
    >
      undefined
    </span>
    <input
      id="dialog-noHTML"
      placeholder="undefined"
    />
  </label>,
}
`;
